src/main/java/com/ericsson/ei/queryservice/ProcessQueryParams.java

```diff
@@ -0,0 +1,168 @@
+        }
+        LOGGER.info("resultAggregatedObject : " + resultAggregatedObject.toString());
+        LOGGER.info("resultMissedNotification : " + resultMissedNotification.toString());
+
+        JSONArray result = null;
+        try {
+            result = ProcessQueryParams.concatArray(resultAggregatedObject, resultMissedNotification);
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+        LOGGER.info("Final Result is : " + result.toString());
+        return result;
+    }
+
+    /**
+     * This method is responsible for concatenating two JSONArrays.
+     *
+     * @param firstArray
+     * @param secondArray
+     * @return JSONArray
+     * @throws JSONException
+     */
+    private static JSONArray concatArray(JSONArray firstArray, JSONArray secondArray) throws JSONException {
+        JSONArray result = new JSONArray();
+        for (int i = 0; i < firstArray.length(); i++) {
+            result.put(firstArray.get(i));
+        }
+        for (int i = 0; i < secondArray.length(); i++) {
+            result.put(secondArray.get(i));
+        }
+        return result;
+    }
+
+    /**
+     * This method takes the parameters from the REST GET request query
+     * parameters and process it to create a JsonNode request to query the
+     * Aggregated Objects. If the Aggregated Object matches the condition, then
+     * it is returned.
+     *
+     * @param request
+     * @return JSONArray
+     */
+    public JSONArray filterQueryParam(String request) {
+        LOGGER.info("The query string is : " + request);
+
+        ObjectMapper mapper = new ObjectMapper();
+        ObjectNode criteria = mapper.createObjectNode();
+        String[] criterias = request.split(",");
+        LOGGER.info("The query parameters are :");
+
+        for (String s : criterias) {
+            String[] node = s.split(":");
+            String key = node[0];
+            String value = node[1];
+            LOGGER.info("The key is : " + key);
+            LOGGER.info("The value is : " + value);
+            criteria.put(key, value);
+        }
+        LOGGER.info(criteria.toString());
+
+        JSONArray resultAggregatedObject = processAggregatedObject.processQueryAggregatedObject(criteria, dataBaseName, aggregationCollectionName);
+        JSONArray resultMissedNotification = processMissedNotification.processQueryMissedNotification(criteria, missedNotificationDataBaseName, missedNotificationCollectionName);
+
+        LOGGER.info("resultAggregatedObject : " + resultAggregatedObject.toString());
+        LOGGER.info("resultMissedNotification : " + resultMissedNotification.toString());
+
+        JSONArray result = null;
+        try {
+            result = ProcessQueryParams.concatArray(resultAggregatedObject, resultMissedNotification);
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage());
+        }
+        LOGGER.info("Final Result is : " + result.toString());
```

Comment:
Code from row 144 to row 157 needs to be extracted in a separate method. This method can also be use in filterFormParam above.